client/src/app/site/motions/services/linenumbering.service.ts

```diff
@@ -0,0 +1,1005 @@
+import { Injectable } from '@angular/core';
+
+const ELEMENT_NODE = 1;
+const TEXT_NODE = 3;
+
+/**
+ * Specifies a point within a HTML Text Node where a line break might be possible, if the following word
+ * exceeds the maximum line length.
+ */
+interface BreakablePoint {
+    /**
+     * The Text node which is a candidate to be split into two.
+     */
+    node: Node;
+    /**
+     * The exact offset of the found breakable point.
+     */
+    offset: number;
+}
+
+/**
+ * An object specifying a range of line numbers.
+ */
+interface LineNumberRange {
+    /**
+     * The first line number to be included.
+     */
+    from: number;
+    /**
+     * The end line number.
+     * HINT: As this object is usually referring to actual line numbers, not lines,
+     * the line starting by `to` is not included in the extracted content anymore, only the text between `from` and `to`.
+     */
+    to: number;
+}
+
+/**
+ * Specifies a heading element (H1, H2, H3, H4, H5, H6) within a HTML document.
+ */
+interface SectionHeading {
+    /**
+     * The first line number of this element.
+     */
+    lineNumber: number;
+    /**
+     * The nesting level. H1 = 1, H2 = 2, etc.
+     */
+    level: number;
+    /**
+     * The text content of this heading.
+     */
+    text: string;
+}
+
+/**
+ * Functionality regarding adding and removing line numbers and highlighting single lines.
```

Comment:
(Should have been in the other review, guess I forgot to click save)

Please provide examples how to use the services.

```
 * ## Example:
 *
 * To add line numbering to a given HTML code:
 *
 * this.lineNumbering.[What to use ]
 * (...)
```

Compare:
https://github.com/OpenSlides/OpenSlides/blob/OpenSlides-3/client/src/app/core/services/viewport.service.ts#L9
https://github.com/OpenSlides/OpenSlides/blob/OpenSlides-3/client/src/app/shared/directives/perms.directive.ts#L10
https://github.com/OpenSlides/OpenSlides/blob/OpenSlides-3/client/src/app/core/services/data-store.service.ts#L200
https://github.com/OpenSlides/OpenSlides/blob/OpenSlides-3/client/src/app/core/services/constants.service.ts#L17
https://github.com/OpenSlides/OpenSlides/blob/OpenSlides-3/client/src/app/core/services/config.service.ts#L18